art_node.go~

```diff
@@ -0,0 +1,682 @@
+	limit := min(len(n.key), len(other.key)) - depth
+
+	i := 0
+	for ; i < limit; i++ {
+		if n.key[depth+i] != other.key[depth+i] {
+			return i
+		}
+	}
+	return i
+}
+
+// Returns the minimum number of children for the current node.
+func (n *ArtNode) MinSize() int {
+	switch n.nodeType {
+	case NODE4:
+		return NODE4MIN
+	case NODE16:
+		return NODE16MIN
+	case NODE48:
+		return NODE48MIN
+	case NODE256:
+		return NODE256MIN
+	default:
+	}
+	return 0
+}
+
+// Returns the maximum number of children for the current node.
+func (n *ArtNode) MaxSize() int {
+	switch n.nodeType {
+	case NODE4:
+		return NODE4MAX
+	case NODE16:
+		return NODE16MAX
+	case NODE48:
+		return NODE48MAX
+	case NODE256:
+		return NODE256MAX
+	default:
+	}
+	return 0
+}
+
+// Returns the Minimum child at the current node.
+// The minimum child is determined by recursively traversing down the tree
+// by selecting the smallest possible byte in each child
+// until a leaf has been reached.
+func (n *ArtNode) Minimum() *ArtNode {
+	if n == nil {
+		return nil
+	}
+
+	switch n.nodeType {
+	case LEAF:
+		return n
+
+	case NODE4, NODE16:
+		return n.children[0].Minimum()
+
+	case NODE48:
+		i := 0
+
+		for n.keys[i] == 0 {
+			i++
+		}
+
+		child := n.children[n.keys[i]-1]
+
+		return child.Minimum()
+
+	case NODE256:
+		i := 0
+		for n.children[i] == nil {
+			i++
+		}
+		return n.children[i].Minimum()
+
+	default:
+	}
+
+	return n
+}
+
+// Returns the Maximum child at the current node.
+// The maximum child is determined by recursively traversing down the tree
+// by selecting the biggest possible byte in each child
+// until a leaf has been reached.
+func (n *ArtNode) Maximum() *ArtNode {
+	if n == nil {
+		return nil
+	}
+
+	switch n.nodeType {
+	case LEAF:
+		return n
+
+	case NODE4, NODE16:
+		return n.children[n.size-1].Maximum()
+
+	case NODE48:
+		i := len(n.keys) - 1
+		for n.keys[i] == 0 {
+			i--
+		}
+
+		child := n.children[n.keys[i]-1]
+		return child.Maximum()
+
+	case NODE256:
+		i := len(n.children) - 1
+		for i > 0 && n.children[byte(i)] == nil {
+			i--
+		}
+
+		return n.children[i].Maximum()
+
+	default:
+	}
+
+	return n
+}
+
+// Replaces the current node with the passed in ArtNode.
+func (n *ArtNode) replaceWith(other *ArtNode) {
+	*n = *other
+}
+
+// Copies the prefix and size metadata from the passed in ArtNode
+// to the current node.
+func (n *ArtNode) copyMeta(other *ArtNode) {
+	n.size = other.size
+	n.prefix = other.prefix
+	n.prefixLen = other.prefixLen
+}
+
+// Key() returns the key used when the node was inserted
+func (n *ArtNode) Key() byte[] {
+	if n.nodeType != LEAF {
+		return nil
+	}
+
+	return n.completeKey
+}
+
+// Returns the value of the given node, or nil if it is not a leaf.
+func (n *ArtNode) Value() interface{} {
+	if n.nodeType != LEAF {
+		return nil
+	}
+
+	return n.value
+}
+
+// Returns the smallest of the two passed in integers.
+func min(a int, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
```

Comment:
Looks like this got caught as a new file (`art_node.go~`), is it perhaps a temporary file that got commited by accident?